packages/server/src/sequencer/sequencer-leader.ts

```diff
@@ -0,0 +1,66 @@
+import { Leader } from '../db/leader'
+import Database from '../db'
+import { sql } from 'kysely'
+import { PLC_SEQ_SEQUENCE } from '../db/types'
+
+const SEQUENCER_LEADER_ID = 1100
+
+export type SequencerLeaderOptions = {
+  pollIntervalMs?: number
+}
+
+export class SequencerLeader {
+  leader: Leader
+  destroyed = false
+  pollIntervalMs: number
+
+  constructor(public db: Database, opts: SequencerLeaderOptions = {}) {
+    this.leader = new Leader(SEQUENCER_LEADER_ID, db)
+    this.pollIntervalMs = opts.pollIntervalMs ?? 50
+  }
+
+  async run(): Promise<{ ran: boolean }> {
+    return this.leader.run(async ({ signal }) => {
```

Comment:
i think this needs to be wrapped in a `while` loop. otherwise if it loses its postgres connection, it will never try to reclaim being a leader again